### ISSUE TITLE
False positive errors when using .append()

### ISSUE THREAD

**Issue by Peterabsolon** [frigus02/typescript-sql-tagged-template-plugin#18]
This code is taken from the documentation. It throws TS errors. How should this be dealt with?

![image](https://user-images.githubusercontent.com/5757143/103168308-2e764680-4832-11eb-810f-8f75a121457a.png)

**Comment by frigus02** [comment]
Hi Peter. That's a good question. It's kind of related to #11.

The plugin parses each template literal on it's own. I think we could in theory construct the full SQL out of multiple template literals. But that would require executing code or at least understand it well enough to know which literals belong together, which seems like a _lot_ of work.

So the short answer is: I don't know.

You could structure your code differently to only have full SQL statements inside a single template literal:

```ts
const query = params.name
    ? sql`SELECT * FROM books WHERE name = ${params.name} LIMIT 10 OFFSET ${params.offset || 0}`
    : sql`SELECT * FROM books LIMIT 10 OFFSET ${params.offset || 0}`
```

There are some benefits to this. E.g. it's easier to copy-paste the entire statement to use it outside the code. But that's not always true and I understand that you might not want to do that.

I can think of a few solutions for this that do not require us to understand the JavaScript/TypeScript code. But I feel like they are either not good enough or too much work for me to implement:

- We could swap out the SQL parser to one that supports partial statements. Does a parser like this exist?

- We could add some kind of comment syntax to help the plugin construct full statements. But that makes reading and writing code harder. E.g.:
  
  ```ts
  sql /* SELECT * FROM foo */ ` WHERE name = ${params.name}`
  ```

So yeah. If you have ideas, I'd love to hear them 🙂

**Comment by Peterabsolon** [comment]
@frigus02 I'm mainly looking to get syntax highlighting from this. I'm commenting here, because afaik, this plugin gets included automatically in VSCode if I use https://www.npmjs.com/package/sql-template-strings

I see you also made a plugin that only does syntax highlighting, but I can't seem to be able to replace the old one with this.

**Comment by Peterabsolon** [comment]
@frigus02 Thanks for the reply. I'm mainly looking to get syntax highlighting from this. I already found your other plugin (thanks!) that does only that so my issue is resolved.

I'm afraid I don't have any ideas on how to solve the underlying issue, my apologies.

**Comment by frigus02** [comment]
Ah, good to hear. I think SQL formatting should work with partial statements. If you want that, you can also disable those statement checks. See `enableDiagnostics` in https://github.com/frigus02/typescript-sql-tagged-template-plugin#configuration
